github/data_source_github_repository.go

```diff
@@ -205,6 +207,13 @@ func dataSourceGithubRepositoryRead(d *schema.ResourceData, meta interface{}) er
 	log.Printf("[DEBUG] Reading GitHub repository %s/%s", owner, repoName)
 	repo, _, err := client.Repositories.Get(context.TODO(), owner, repoName)
 	if err != nil {
+		if err, ok := err.(*github.ErrorResponse); ok {
+			if err.Response.StatusCode == http.StatusNotFound {
+				log.Printf("Error reading GitHub branch reference %s/%s: %s", owner, repoName, err)
```

Comment:
What do you thinking about giving this a log level of `[DEBUG]` similar to line 207 above?

Reply:
That makes sense. I took this from the branch equivalent, so I'll go ahead and update both places. Also, the message isn't correct. I'll change it to better reflect the 404 that we know is the cause here.